src/methods.ts

```diff
@@ -158,10 +159,15 @@ export async function upload(inputs: InputsArtifacts): Promise<void> {
   }
   const globber = await glob.create(inputs.source, globOptions)
   for await (const file of globber.globGenerator()) {
-    requests.push(upload_one_file(4, file, dirname, name, inputs))
+    requests.push(file)
   }
   core.info('Waiting for all requests to finish')
-  await Promise.all(requests)
+
+  function mapFile(file: string): void {
+    upload_one_file(4, file, dirname, name, inputs)
+  }
+  async.mapLimit(requests, 16, mapFile);
```

Comment:
Are we missing an await here? Or does the function does that on it's own?
```suggestion
  await async.mapLimit(requests, 16, mapFile);
```